node/resolvers/directives/withPermissions.ts

```diff
@@ -21,21 +30,18 @@ export class WithPermissions extends SchemaDirectiveVisitor {
 
       const appClients = context.vtex as any
 
-      appClients.storefrontPermissions = await storefrontPermissions
-        .checkUserPermission()
-        .then((result: any) => {
-          return result?.data?.checkUserPermission ?? null
-        })
-        .catch((error: any) => {
-          if (!adminUserAuthToken) {
-            logger.error({
-              message: 'getPermissionsError',
-              error,
-            })
-          }
-
-          return null
-        })
+      appClients.storefrontPermissions = await getCheckUserPermission(
+        storefrontPermissions
+      ).catch((error: any) => {
+        if (!adminUserAuthToken) {
+          logger.error({
+            message: 'getPermissionsError',
+            error,
+          })
+        }
+
+        return null
+      })
```

Comment:
I won't ask you to change this one, as it was already here before. But just to further explain, this code here is equivalent to:

```suggestion
      try {
        appClients.storefrontPermissions = await getCheckUserPermission(
          storefrontPermissions
        )
      } catch (error) {
        if (!adminUserAuthToken) {
          logger.error({
            message: 'getPermissionsError',
            error,
          })
        }

        // The return value from `.catch()` promise functions is the value returned by the promise,
        // so it'd be set to `appClients.storefrontPermissions` in the original code.
        appClients.storefrontPermissions = null
      }
```